hphp/runtime/vm/jit/vasm-unit.h

```diff
@@ -211,6 +211,7 @@ struct Vunit {
   uint16_t cur_voff{0};  // current instruction index managed by Vout
   bool padding{false};
   bool profiling{false};
+  bool avoidVreg{false}; // Indication to lowerers to avoid use of new Vregs.
```

Comment:
Yeah I definitely do too, and I like you approach to implementing those assertions. I still can't think of how to get around the portion of decRefHelper that can safely use virtuals. Initially I just started adding the same support in those lowerers, but I hit a wall in some of them. In particular, I ran into some complicated Vptrs that made use of both a base and index. When lowering, two new virtuals are required and their live ranges overlap. I ran out of steam at that point. I have plans to streamline lowerVptr quite a bit, and that might accommodate lowering without using new Vregs.

With that said, I still prefer to actually figure out a way to simple relax the `allow_implicit_vregs` when emitting code protected by a PhysRegSaver because it *should* lead to better code that balances register use more on platforms that don't support register renaming (Cortex-A53 for instance).

Let me know if you think we can postpone the asserts, and I'm about to address your comment about the flag name now.